--- kmod-34.1/libkmod/libkmod.c.orig	2025-03-06 06:25:07.000000000 +0100
+++ kmod-34.1/libkmod/libkmod.c	2025-03-20 19:46:31.547435280 +0100
@@ -41,8 +41,10 @@ static const struct {
 	// clang-format on
 };
 
+static char kver_config_path[PATH_MAX] = SYSCONFDIR "/modprobe.d/X.Y.Z";
 static const char *const default_config_paths[] = {
 	// clang-format off
+	kver_config_path,
 	SYSCONFDIR "/modprobe.d",
 	"/run/modprobe.d",
 	"/usr/local/lib/modprobe.d",
@@ -242,8 +244,16 @@ KMOD_EXPORT struct kmod_ctx *kmod_new(co
 
 	ctx->kernel_compression = get_kernel_compression(ctx);
 
-	if (config_paths == NULL)
-		config_paths = default_config_paths;
+	if (config_paths == NULL) {
+		struct utsname u;
+
+		if (uname(&u) < 0) {
+			config_paths = &default_config_paths[1];
+		} else {
+			snprintf(kver_config_path, PATH_MAX, SYSCONFDIR "/modprobe.d/%s", u.release);
+			config_paths = default_config_paths;
+		}
+	}
 	err = kmod_config_new(ctx, &ctx->config, config_paths);
 	if (err < 0) {
 		ERR(ctx, "could not create config\n");
