--- kmod-26/configure.ac~	2019-02-07 22:41:45.000000000 +0100
+++ kmod-26/configure.ac	2019-11-02 20:42:34.219987092 +0100
@@ -176,7 +176,7 @@
 	AM_PATH_PYTHON(,,[:])
 	AC_PATH_PROG([CYTHON], [cython], [:])
 
-	PKG_CHECK_MODULES([PYTHON], [python-${PYTHON_VERSION}],
+	PKG_CHECK_MODULES([PYTHON], [python-${PYTHON_VERSION}-embed],
 			  [have_python=yes],
 			  [PKG_CHECK_MODULES([PYTHON], [python],
 					     [have_python=yes],
